jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler1.java

```diff
@@ -0,0 +1,96 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server.handler;
+
+import org.eclipse.jetty.http.pathmap.PathSpecSet;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.gzip.GzipHandler;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.IncludeExclude;
+
+public abstract class ConditionalHandler1 extends Handler.Wrapper
+{
+    @Override
+    public boolean handle(Request request, Response response, Callback callback) throws Exception
+    {
+        if (canHandle(request))
+            return super.handle(request, response, callback);
+
+        // Skip any composed conditional handlers
+        Handler handler = getHandler();
+        while (handler instanceof ConditionalHandler1 conditional)
```

Comment:
I like the concept of the `canHandle(request)` but don't like how we unwrap the other `ConditionalHandler1`s.

Could we do a version of this handler which takes a list of `Predicate<Request>` and uses that to decide what to do. Then you can add predicates for `Paths`, `Methods`, `InetAccessSet` or whatever other different use case we have in the future.

Reply:
@lachlan-roberts See the commit I just pushed,  as that is kind of a step towards what you are suggesting.   But instead of a list of predicates, there is an IncludeExcludeSet of tuples.
That could be extended to be an IncludeExcludeSet of predicates.

However, we are now going to iterate over all these looking for matches, which could be inefficient.

Reply:
@lachlan-roberts to be more specific, the advantage of being able to use something like `new IncludeExclude<>(PathSpecSet.class);` is that lookup can be non-linear as maps and tries can be used internally.
Both the list of predicates and the InetAccessHandler approach do not allow this.

Perhaps we use the MethodHandler approach of ConditionalHandler4 and allow a set of general predicates to be added, which if empty will not be called at all.

Reply:
@lachlan-roberts see the latest, which now supports an arbitrary include/exclude of predicates.